posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc

```diff
@@ -0,0 +1,413 @@
+
+*	IBM Cloud Code Engine: IBM Cloud Function
+*	AWS: AWS Lambda
+*	Microsoft Azure: Azure Functions
+*	Google Cloud: Google Cloud Functions
+
+
+== CloudEvents
+https://knative.dev/docs/concepts[Knative] and https://cloudevents.io/[CloudEvents] are  two frameworks that are often used to work with the serverless model. This post focuses on CloudEvents. 
+
+CloudEvents is a specification to describe event data in a common language-agnostic way. CloudEvents simplifies event declaration and delivery across the services and platforms. This specification is under the Cloud Native Computing Foundation (CNCF). CloudEvents are supported by many serverless frameworks, including IBM Cloud Code Engine.
+
+== IBM Cloud Code Engine
+https://www.ibm.com/cloud/code-engine[IBM Cloud Code Engine](ICCE) is a fully managed, serverless platform designed to run container images, batch jobs, or source code. Code Engine allows you to build event driven workloads that react to CloudEvents. Code Engine is truly serverless. It automatically scales your workloads up and down, even down to zero when there are no requests. You pay for only the resources you consume. You can easily run your cloud native applications in Code Engine. 
+
+== MicroProfile Applications
+Many developers use MicroProfile specifications for configuring, securing, and observing cloud native applications. MicroProfile offers a set of standards APIs for cloud-native applications that free your them from vendor lock-in. Open Liberty is the leading implementation for MicroProfile specifications.
+
+As we enter the serverless era, it is very important to get MicroProfile applications running in a serverless environment. Luckily, it is very straightforward to get these applications running as functions in an environment that supports CloudEvents, such as IBM Cloud Code Engine. In the following sections, we demonstrate how to get a MicroProfile application that is deployed to Open Liberty running as a function in IBM Cloud Code Engine with CloudEvent.
+
+== Run MicroProfile applications  as a serverless function in Code Engine
+RESTful MicroProfile Applications can run seamlessly as a serverless function in Code Engine with CloudEvent. This section explains how.
+
+=== Prerequisites
+Before you start you will need the following prerequisites:
+
+* An IBM Cloud Account
+* Docker
+* An externally accessible container Registry that can host your images (IBM Cloud Container Registry is available as part of your cloud account)
+
+=== Creating your MicroProfile application
+If you don't have a suitable RESTful application to host as a set of functions you can use the https://openliberty.io/guides/getting-started.html[OpenLiberty Getting Started guide] to get a simple MicroProfile application that is  well suited to a serverless environment, such as IBM Cloud Code Engine.
+
```

Comment:
consider just using the getting started guide for this post- trying to allow for someone to byo app makes the task flow more complicated, and we don't really give enough info. If someone wants to use another app, they'd need to understand well enough how to do it anyway. You can mention that it would work with any RESTful application but still focus the task and description just on using the Getting Started guide app.

Reply:
agreed. Just update it: In this blog post, we will use the Getting Started Guide.

Reply:
updated